### PR TITLE
Really get track

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #5772

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: libraries/lib-wave-track/WaveTrack.cpp [hunk 1]
```diff
@@ -3439,13 +3472,12 @@ std::pair<float, float> WaveChannel::GetMinMax(
    if (t0 == t1)
       return results;
 
-   for (const auto &clip: GetTrack().mClips)
+   for (const auto &&clip: Intervals())
```
Comment (crsib) [inline libraries/lib-wave-track/WaveTrack.cpp:3475]: Constant rvalue references are highly unusual. Why they are needed here?

Comment (Paul-Licameli) [reply libraries/lib-wave-track/WaveTrack.cpp:3475]: I will strike const

Comment (crsib) [reply libraries/lib-wave-track/WaveTrack.cpp:3475]: Why the rvalue reference is needed here?

Comment (Paul-Licameli) [reply libraries/lib-wave-track/WaveTrack.cpp:3475]: Const was a mistake.

&& isn’t necessary but it is a reminder that the interval iterator dereferences to a value, not a reference as in the old version.

Interval iterators dereference to a shared pointer to a temporary structure.

**Comment by Paul-Licameli** [comment]
Rebased onto the new dependency, and also fixed that rvalue references in loops.

**Pull request merged**